feat: add verbosity param to chain related rpc

quake opened the pull request:
This PR adds an optional `verbosity ` param to chain related rpc, returns data in hex format without calculated hash values, it will improve performance in some scenarios, ~~for example, ckb-indexer can scan 2,000,000 blocks in 13 minutes with this change (before: 20 minutes)~~ **This benchmark is not correct, I will rerun it and update**

=== updated ckb-indexer benchmark result ===
scan 2,000,000 blocks, run 3 times, avg:
before: 17 minutes 40 seconds
after: 12 minutes 30 seconds


=== response example ===
hex format: 
```
echo '{                                                                                     
    "id": 2,     
    "jsonrpc": "2.0",
    "method": "get_tip_header",
    "params": ["0x1"]
}' \
| tr -d '\n' \
| curl -H 'content-type: application/json' -d @- \
http://localhost:8114

{"jsonrpc":"2.0","result":"0x000000002872ae19e93f872d7401000027c92700000000009e06003d02080700903e152458ff060ddf290e2b876d94211be8982f4b10ba63b9b017abb587bb75f59cd510ec3e3e13ff331a41d954f3ed9b9980cdcea5c88b5e7bf6fcfcd2bc2c000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000005ed103b7d5f09434388f10dcba93240032b96b8ee42e3a01001579760524020700043bd6280000002b99f9d300010000","id":2}
```

v.s

json format
```
echo '{
    "id": 2,
    "jsonrpc": "2.0",
    "method": "get_tip_header",
    "params": []    
}' \
| tr -d '\n' \
| curl -H 'content-type: application/json' -d @- \
http://localhost:8114
{"jsonrpc":"2.0","result":{"compact_target":"0x19ae7228","dao":"0x5ed103b7d5f09434388f10dcba93240032b96b8ee42e3a010015797605240207","epoch":"0x708023d00069e","hash":"0x50f0a70e5fdbce353a46f018c1a6f1dd15cb7470fff8ebcb2b7b418cd1a5faab","nonce":"0x100d3f9992b00000028d63b0400","number":"0x27c927","parent_hash":"0x903e152458ff060ddf290e2b876d94211be8982f4b10ba63b9b017abb587bb75","proposals_hash":"0x0000000000000000000000000000000000000000000000000000000000000000","timestamp":"0x1742d873fe9","transactions_root":"0xf59cd510ec3e3e13ff331a41d954f3ed9b9980cdcea5c88b5e7bf6fcfcd2bc2c","uncles_hash":"0x0000000000000000000000000000000000000000000000000000000000000000","version":"0x0"},"id":2}
```

zhangsoledad commented:
I suggest using `verbosity`, this has been widely accepted. description comes easier.
https://bitcoin-rpc.github.io/en/doc/0.17.99/rpc/blockchain/getblock/
we may consider return information block only contains tx_hash later.

doitian commented:
Does it return hex data without decoding? The doc “without calculated hash values” is misleading.

doitian commented:
It’s better to give example responses when hex format is choosed

quake commented:
> I suggest using `verbosity` and `verbose`, this has been widely accepted. description comes easier.
> https://bitcoin-rpc.github.io/en/doc/0.17.99/rpc/blockchain/getblock/
> https://bitcoin-rpc.github.io/en/doc/0.17.99/rpc/rawtransactions/getrawtransaction/
> we may consider return information block only contains tx_hash later.

I decided to use an unified numeric type `verbosity` params for all hex encoding rpc, please review again, thanks.

quake commented:
> It’s better to give example responses when hex format is choosed

I've updated the rpc document format, added markdown link in the params description, it will help user to understand the hex encoding format better, however, still cannot write the example hex response in doc because of the format limitation.

doitian commented on rpc/json/rpc.json:
```diff
@@ -24,7 +24,12 @@
+        },
+        "types": [
+            {
+                "verbosity": "0 for a json object, 1 for hex encoded [Header](../util/types/schemas/blockchain.mol#L84) data, an optional parameter, default is 0"
```
Isn't json more verbose than hex? 🤣 

From bitcoin

```
0 for hex encoded data, 1 for a json object, and 2 for json object with transaction data
```

quake replied:
Updated, please review again, thanks.

doitian commented on rpc/README.md:
```diff
@@ -499,8 +511,8 @@ Returns the information about a cell by out_point if it is live. If second with_
-    out_point - OutPoint object {"tx_hash": <tx_hash>, "index": <index>}.
-    with_data - Boolean
+* out_point - OutPoint object {"tx_hash": <tx_hash>, "index": <index>}.
+* with_data - Boolean
```
A candidate to be replaced by `verbosity` later.

quake commented on rpc/json/rpc.json:
```diff
@@ -133,6 +138,9 @@
                 "hash": "Hash of a block"
+            },
+            {
+                "verbosity": "2 for a json object, 0 for hex encoded [Block](../util/types/schemas/blockchain.mol#L94) data, an optional parameter, default is 2"
```
will create a PR later: return block only contains tx_hash in json format and use the verbosity=1

quake commented:
bors r+

quake commented:
bors r=doitian,zhangsoledad
